From 6ce807817b772ea84fc7cfc763ccb6be77fdd900 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:40:26 -0400
Subject: [ethernet] i40e: use correct structure type name in sizeof

Message-id: <1409658042-22689-266-git-send-email-sassmann@redhat.com>
Patchwork-id: 90466
O-Subject: [RHEL7.1 PATCH 265/281] i40e: use correct structure type name in sizeof
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Julia Lawall <Julia.Lawall@lip6.fr>

Correct typo in the name of the type given to sizeof.  Because it is the
size of a pointer that is wanted, the typo has no impact on compilation or
execution.

This problem was found using Coccinelle (http://coccinelle.lip6.fr/).  The
semantic patch used can be found in message 0 of this patch series.

Signed-off-by: Julia Lawall <Julia.Lawall@lip6.fr>
Tested-By: Jim Young <jamesx.m.young@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit f57e4fbdcc8f65b94d59ce5566c781c46a298c28)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index ce30a94..8f33549 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -6293,7 +6293,7 @@ static int i40e_vsi_alloc_arrays(struct i40e_vsi *vsi, bool alloc_qvectors)
 
  if (alloc_qvectors) {
   /* allocate memory for q_vector pointers */
-  size = sizeof(struct i40e_q_vectors *) * vsi->num_q_vectors;
+  size = sizeof(struct i40e_q_vector *) * vsi->num_q_vectors;
   vsi->q_vectors = kzalloc(size, GFP_KERNEL);
   if (!vsi->q_vectors) {
    ret = -ENOMEM;
-- 
1.7.1